pkg/skaffold/runner/dev.go

```diff
@@ -21,17 +21,36 @@ import (
 	"io"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/color"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/filemon"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/portforward"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/sync"
-	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
 // ErrorConfigurationChanged is a special error that's returned when the skaffold configuration was changed.
 var ErrorConfigurationChanged = errors.New("configuration changed")
 
+func (r *SkaffoldRunner) needsAction() bool {
+	return r.changeSet.needsReload || len(r.changeSet.needsRebuild) > 0 || r.changeSet.needsRedeploy
+}
+
+func (r *SkaffoldRunner) processArtifacts() error {
```

Comment:
I'm not sure extracting this piece of code makes it easier to read. It seems disconnected to the rest of the code now. If you still prefer to extract it, at least maybe change the name.

Reply:
the main reason I did this is because it's also being used in the tests, and it felt better than having duplicated code. I'll leave it moved out for now, but I'll put a TODO to try and move it into the callback registered with the monitor. I changed the name to `separateBuildAndSync()`, suggestions welcome though (I don't like this one either)